### PR TITLE
195 frontend workout frequency chart 1

### PR DESCRIPTION
**Pull request by BitsyBirb**
# What type of PR is this? (Check all that apply)

- [X] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

What did you change? How did you change it?

Implemented line chart that plots how many times the logged-in user has exercised in each month of the current year.

Component can be found in `~/apps/expo/src/components/workout/frequency_graph.tsx` and the name of the component itself is `DisplayWorkoutFrequencyGraph({userID: number})`. You pass the userID of the global context into it.

Has styling issues that I'd like to address with the rest of the team such as finalizing the design or adding to it, but for now the prototype is good to go and gets the proper data/plots it.

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [X] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots
image

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [X] NO documentation needed

Link to external documentation: https://www.npmjs.com/package/react-native-gifted-charts

# [Optional] Are there any post-deployment tasks we need to perform?

Besides reviewing the styling and cleaning it up a bit, not to my understanding.

### CODE REVIEW

File: apps/expo/package.json [hunk 1]
```diff
@@ -51,10 +51,13 @@
     "react-dom": "18.2.0",
     "react-native": "0.73.6",
     "react-native-body-highlighter": "3.0.0-beta.6",
+    "react-native-chart-kit": "^6.12.0",
```
Comment (BitsyBirb) [inline apps/expo/package.json:54]: No longer needed for my chart. Can remove.

**Review by StefanoRubini — APPROVED** [review]
Thank you Miguel for creating this workout frequency chart for us. After running it on my local machine, I can verify that the graph is working as intended and looks the same as the one provided in the screenshot. Great work! 😃

File: apps/expo/package.json (hunk 1 above)
Comment (thienguen) [reply apps/expo/package.json:54]: can you remove the extra branch as well

Comment (BitsyBirb) [reply apps/expo/package.json:54]: Extra branch?

**Review by taylorfinelli — CHANGES_REQUESTED** [review]
Can you try to make it look more like the mock? 

Screenshot 2024-04-24 at 3 03 47 PM

**Comment by taylorfinelli** [comment]
If you need sample code to look at for the styling you can look at `apps/expo/src/components/post/PostDashboard.tsx`

File: apps/expo/package.json (hunk 1 above)
Comment (thienguen) [reply apps/expo/package.json:54]: yes, extra branch

File: apps/expo/src/components/workout/frequency_graph.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import {LineChart} from 'react-native-gifted-charts'
+
+// To get the user's data, in particular the WorkoutLogs
+import { api } from '~/utils/trpc/api'
+import RotatingBarbellIcon from '../notif/RotatingBarbellIcon'
+
+
+// Just use react-native-chart-kit to display the user's workout across each month for current year.
+export default function DisplayWorkoutFrequencyGraph({userID} : {userID: number}){
+    let values = [
```
Comment (taylorfinelli) [inline apps/expo/src/components/workout/frequency_graph.tsx:15]: is this mock data?

File: apps/expo/package.json (hunk 1 above)
Comment (thienguen) [reply apps/expo/package.json:54]: Screenshot_20240424-150535

**Comment by mva919** [comment]
I think you can style it as the mock up if you just change the background color

File: apps/expo/src/components/workout/frequency_graph.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+
+    return(
+        <View>{isFetching ? <View><RotatingBarbellIcon /></View> : 
+            <>
+            <Text>Line Chart for User: {userID}</Text>
+            <LineChart 
+                rotateLabel={true}
+                curved
```
Comment (taylorfinelli) [inline apps/expo/src/components/workout/frequency_graph.tsx:60]: can we make the lines straight instead of curved

Comment (BitsyBirb) [reply apps/expo/src/components/workout/frequency_graph.tsx:60]: Yah, I can just quickly remove that curved property.

**Comment by taylorfinelli** [comment]
Added myself and marcos as reviewers

File: apps/expo/src/components/workout/frequency_graph.tsx (hunk 2 above)
Comment (BitsyBirb) [reply apps/expo/src/components/workout/frequency_graph.tsx:15]: The values array is necessary, but I can zero out the rest of the array until it gets populated if that's what you were asking. I can look into finding a way to truncate the graph but not sure if that would mess up sizing.

**Comment by taylorfinelli** [comment]
Use case:

Screenshot 2024-04-24 at 3 09 04 PM

**Comment by taylorfinelli** [comment]
Updated: 

Screenshot 2024-04-28 at 5 28 39 PM

File: apps/expo/src/components/workout/frequency_graph.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    <View style={{ paddingTop: 7 }}>
+      <LineChart
+        data={values}
+        startFillColor='#FFFFFF'
+        thickness={2}
+        color={'#FFFFFF'}
+        backgroundColor={'#48476D'}
+        dataPointsColor={'#FFFFFF'}
+        dataPointsRadius={3}
+        spacing={screenWidth * 0.09}
```
Comment (taylorfinelli) [inline apps/expo/src/components/workout/frequency_graph.tsx:86]: idk how this scales on smaller devices but it should look fine if we use an IOS device for demo. we can fix after demo if needed

Comment (BitsyBirb) [reply apps/expo/src/components/workout/frequency_graph.tsx:86]: Yeah; it looked fine on Android as well (the graph at least).

File: apps/expo/package.json (hunk 1 above)
Comment (BitsyBirb) [reply apps/expo/package.json:54]: Done I believe.

**Review by mva919 — APPROVED** [review]
good job thanks for fixing the styling

**Pull request merged**